### ISSUE TITLE
Add the ability to delay ScreenshotAsync

### ISSUE THREAD

**Issue by zandraswo** [microsoft/playwright#28757]
I have tried using ScreenshotAsync method and got disappointed because of its inconsistency. A different result is produced due to animations. I have tried to use "Animations = ScreenshotAnimations.Disabled" option but it does not seem to work.

When I tested adjusting SlowMo setting during test execution, I found that this made the results noticeably more consistent:

- I did 10  test runs taking the same screenshot and the accuracy was 40-60%.
- After setting SlowMo to 2000 ms, 10/10 runs were identical.

I believe adding a customizable delay to ScreenshotAsync method would:

- Increase the reliability of the method.
- Allow to reduce the screenshot speed without affecting the whole suite using SlowMo.
- Allow to target elements that take longer to load than others, e.g., graphs with lots of data.

**Comment by yury-s** [comment]
How is this different from adding a random delay before calling `ScreenshotAsync`? Relying on such delay would be inherently flaky. We instead recommend using [web assertions](https://playwright.dev/docs/best-practices#use-web-first-assertions) to wait until the page gets into a desired [user visible](https://playwright.dev/docs/best-practices#test-user-visible-behavior) state and take screenshot after that.

**Comment by zandraswo** [comment]
> How is this different from adding a random delay before calling `ScreenshotAsync`? Relying on such delay would be inherently flaky. We instead recommend using [web assertions](https://playwright.dev/docs/best-practices#use-web-first-assertions) to wait until the page gets into a desired [user visible](https://playwright.dev/docs/best-practices#test-user-visible-behavior) state and take screenshot after that.

Thanks for the suggestion. After some attempts I managed to find a way to solve this using web assertions. The problem was a bit more interesting, so I figured to share what happened in case someone runs in to a similar issue.

The original code was doing this:

1. Open a page.
2. Click on a button.
3. Wait for the element to load.
4. Take a screenshot.

The problem was that I was waiting at the wrong place. Clicking on a button too quickly actually affected what happened next due to some inner workings of the underlying web framework.

The code had to be adjusted to:

1. Open a page.
2. Wait for certain elements to load.
3. Click on a button (not that I didn't even need to wait at this step anymore).
4. Take a screenshot.
